localflavor/nl/models.py

```diff
@@ -1,62 +1,88 @@
-import re
-from django.core.exceptions import ValidationError
-from django.core.validators import RegexValidator
+from __future__ import absolute_import, unicode_literals
+
 from django.db import models
 from django.utils.translation import ugettext_lazy as _
 
+from . import forms
+from .nl_provinces import PROVINCE_CHOICES
+from .validators import (NLBankAccountNumberFieldValidator,
+                         NLPhoneNumberFieldValidator,
+                         NLSoFiNumberFieldValidator, NLZipCodeFieldValidator)
+
 
-class NLBankAccountNumberFieldValidator(RegexValidator):
+class NLZipCodeField(models.CharField):
+    """
+    A Dutch zip code model field
     """
-    Validation for Dutch bank accounts.
+    description = _('Dutch zipcode')
 
-    Validation references:
-    http://www.mobilefish.com/services/elfproef/elfproef.php
-    http://www.credit-card.be/BankAccount/ValidationRules.htm#NL_Validation
+    validators = [NLZipCodeFieldValidator()]
 
-    .. versionadded:: 1.1
-    """
-    default_error_messages = {
-        'invalid': _('Enter a valid bank account number'),
-        'wrong_length': _('Bank account numbers have 1 - 7, 9 or 10 digits'),
-    }
+    def __init__(self, *args, **kwargs):
+        kwargs['max_length'] = 7
+        super(NLZipCodeField, self).__init__(*args, **kwargs)
 
-    def __init__(self, regex=None, message=None, code=None):
-        super(NLBankAccountNumberFieldValidator, self).__init__(regex='^[0-9]+$',
-                                                                message=self.default_error_messages['invalid'])
-        self.no_leading_zeros_regex = re.compile('[1-9]+')
+    def to_python(self, value):
+        return '%s %s' % (value[:4], value[4:])
```

Comment:
I found a problem here. This is adding an extra space when the model field is used with the form field. For example, if I enter '1000AA', I will get the error message: 'Ensure this value has at most 7 characters (it has 8).' because the form adds a space and then this adds a space. You'll need to check if the space is already there before adding it. You should also add a test for this. Thanks.